src/main/java/org/folio/circulation/domain/policy/LoanPolicy.java

```diff
@@ -315,7 +315,7 @@ private boolean isProfile(JsonObject loansPolicy, String profileId) {
 
   LoanPolicy withDueDateSchedules(FixedDueDateSchedules loanSchedules) {
     return new LoanPolicy(representation, loanSchedules,
-      alternateRenewalFixedDueDateSchedules);
+      alternateRenewalFixedDueDateSchedules, ruleConditionsEntity);
```

Comment:
What is a `ruleConditionsEntity`?

Should it be required for all instances of `LoanPolicy`?

Reply:
ruleConditionsEntity is an object which we use for itemLimit checking. It contains 3 fields:   - boolean isItemTypePresent;
boolean isLoanTypePresent;
boolean isPatronGroupPresent;
That indicates which conditions were applied for matching circulation rule.

Reply:
> Should it be required for all instances of LoanPolicy?

Yes, it should. For this case with fixed due date schedule, new LoanPolicy was created without ruleConditionsEntity. That was a reason why we got NPE in ItemLimitValidator.

Reply:
> > Should it be required for all instances of LoanPolicy?
> 
> Yes, it should. For this case with fixed due date schedule, new LoanPolicy was created without ruleConditionsEntity. That was a reason why we got NPE in ItemLimitValidator.

Please could it be made a required constructor parameter (by removing any overload that does not have it) and possibly disallowing null values for that parameter

Reply:
Thanks, removed